Implement user model with MVC structure, MongoDB integration, and full CRUD

OfekSagiv opened the pull request:
- MVC structure (controllers, services, repositories, models)
- MongoDB connection via Mongoose 7+
- Full CRUD operations for users
- Basic error handling and async/await
- All HTTP methods tested successfully via Postman

@Tamir198

Tamir198 commented on src/app.js:
```diff
@@ -1,9 +1,9 @@
 app.use(express.json());
-app.use('/api/v1', api);
+app.use('/api/v1/users', userRoutes);
 
```
I would create a  router file to remove all of the `app.use(router)` from here, something like this : 

```js
const express = require('express');
const router = express.Router();

const firstRouter = require('./firstRouter');
const secondRouter = require('./secondRouter');

router.use(firstRouter);
router.use(secondRouter);

module.exports = router;

```

Then call this form `app.js`

Tamir198 commented on src/controllers/userController.js:
```diff
@@ -0,0 +1,50 @@
+const deleteUser = async (req, res) => {
+  try {
+    await userService.deleteUser(req.params.id);
+    res.status(204).send();
```
Destruct your `req` object instead

Tamir198 commented on src/models/userModel.js:
```diff
@@ -0,0 +1,44 @@
+  role: {
+    type: String,
+    enum: ['user', 'admin'],
+    default: 'user'
```
Get the values from constant files

Tamir198 commented on src/models/userModel.js:
```diff
@@ -0,0 +1,44 @@
+  }
+}, { timestamps: true });
+
+module.exports = mongoose.model('User', userSchema);
```
Same for User

Tamir198 commented on src/repositories/userRepository.js:
```diff
@@ -0,0 +1,23 @@
+const createUser = (userData) => User.create(userData, {});
+
+const getAllUsers = () => User.find({}, {}, {});
+
```
You don't need all of the parameters of find .

From the documentation of mongoose : 

_find_ all documents
```js
await MyModel.find({});
```

OfekSagiv replied:
I changed it because of the compiler warnings. this is the standard  of new mongoose version. should I remove it ?

Tamir198 commented on src/repositories/userRepository.js:
```diff
@@ -0,0 +1,23 @@
+const getAllUsers = () => User.find({}, {}, {});
+
+const getUserById = (id) => User.findById(id, {}, {});
+
```
Same here you don't need those empty objects. 

![image](https://github.com/user-attachments/assets/8004c83b-4130-40c8-ade5-9f3b81c0bed7)

OfekSagiv replied:
I changed it because of the compiler warnings. this is the standard  of new mongoose version. should I remove it ?
@Tamir198

Tamir198 commented on src/repositories/userRepository.js:
```diff
@@ -0,0 +1,23 @@
+const User = require('../models/userModel');
+
+const createUser = (userData) => User.create(userData, {});
+
```
What is the purpose of the second parameter of the create method? 

Here is an example from the documentation about creating a user:  

![image](https://github.com/user-attachments/assets/d2cd28c5-a559-4de9-89c2-e1dfe89e3416)

Tamir198 commented on src/repositories/userRepository.js:
```diff
@@ -0,0 +1,23 @@
+  });
+
+const deleteUser = (id) => User.findByIdAndDelete(id, {});
+
```
Same comment about the second argument here

Tamir198 commented on src/routes/userRoutes.js:
```diff
@@ -0,0 +1,12 @@
+
+router.get('/', userController.getAllUsers);
+router.get('/:id', userController.getUserById);
+router.post('/', userController.createUser);
```
Save this into a variable/ constant

Tamir198 commented on src/services/userService.js:
```diff
@@ -0,0 +1,37 @@
+const userRepository = require('../repositories/userRepository');
+
+const createUser = async (userData) => userRepository.createUser(userData);
+
```
I will check this file after the changes inside the userRepository

Tamir198 reviewed (changes requested):
I would replace all of the hardcoded status codes with constants, luckiily there is already a library for this : 


https://www.npmjs.com/package/http-status-codes

All of your error messages should wither come from a constant file or a library like the library above. 

For the future we will need to add error handling middleware to sit globally

OfekSagiv commented:
Addressed all feedback: moved constants to files, replaced hardcoded status codes with http-status-codes, simplified repository queries, organized routers, and reverified all endpoints in Postman.
@Tamir198

OfekSagiv commented:
done.

Tamir198 reviewed (approved):
Looks good, you can merge
